FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+
+void * pvPortMalloc( size_t xSize )
+{
+    return unity_malloc( xSize );
+}
+void vPortFree( void * pv )
+{
+    return unity_free( pv );
+}
+
+/* ============================  Unity Fixtures  ============================ */
+/*! called before each testcase */
+void setUp( void )
+{
+    xEventGroupHandle = NULL;
+
+    pxListTemp->pxIndex = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.xItemValue = portMAX_DELAY;
+    pxListTemp->xListEnd.pxNext = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->xListEnd.pxPrevious = ( ListItem_t * ) &( pxListTemp->xListEnd );
+    pxListTemp->uxNumberOfItems = ( UBaseType_t ) 0U;
+
+    pxListItem_HasTaskBlockOnBit0->xItemValue = BIT_0;
+
+    vFakeAssert_Ignore();
+    vFakePortEnterCriticalSection_Ignore();
+    vFakePortExitCriticalSection_Ignore();
+    ulFakePortSetInterruptMaskFromISR_IgnoreAndReturn( 0U );
+    vFakePortClearInterruptMaskFromISR_Ignore();
+
+    /* Track calls to malloc / free */
+    UnityMalloc_StartTest();
+}
+
+/*! called after each testcase */
+void tearDown( void )
+{
+    UnityMalloc_EndTest();
+}
+
+/*! called at the beginning of the whole suite */
+void suiteSetUp()
+{
+}
+
+/*! called at the end of the whole suite */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ===========================  Static Functions  =========================== */
+
+
+
+/* ==============================  Test Cases  ============================== */
+
+/*!
+ * @brief validate dynamically creating and deleting a new RTOS event group,
+ * @coverage xEventGroupCreate
+ */
+void test_xEventGroupDynamicCreateAndDelete_Success( void )
+{
+    /* Expectation of Function: xEventGroupCreate */
+    vListInitialise_Expect( 0 );
+    vListInitialise_IgnoreArg_pxList();
+    vListInitialise_ReturnThruPtr_pxList( pxListTemp );
+    /* Expectation of Function: vEventGroupDelete */
+    vTaskSuspendAll_Ignore();
+    listCURRENT_LIST_LENGTH_ExpectAnyArgsAndReturn( 1 );
+    vTaskRemoveFromUnorderedEventList_Ignore();
+    listCURRENT_LIST_LENGTH_ExpectAnyArgsAndReturn( 0 );
+    xTaskResumeAll_IgnoreAndReturn( 1 );
+
+    /* API to Test */
+    xEventGroupHandle = xEventGroupCreate();
+    TEST_ASSERT_NOT_EQUAL( NULL, xEventGroupHandle );
+
+    vEventGroupDelete( xEventGroupHandle );
+}
+
+
+/*!
+ * @brief validate dynamically creating a event group failed when malloc failed
+ * @coverage xEventGroupCreate
+ */
+void test_xEventGroupDynamicCreate_FailMalloc( void )
+{
+    /* Expectation of Function: xEventGroupCreate */
+    vListInitialise_Ignore();
+    UnityMalloc_MakeMallocFailAfterCount( 0 );
+
+    /* API to Test */
+    xEventGroupHandle = xEventGroupCreate();
+
+    /* Validate */
+    TEST_ASSERT_EQUAL( NULL, xEventGroupHandle );
+}
+
+
+/*!
+ * @brief validate statically creating and deleting a new RTOS event group,
+ * @coverage xEventGroupCreateStatic
+ */
+void test_xEventGroupStaticCreate_Success( void )
+{
```

Comment:
Consider adding a configASSERT to xEventGroupCreateStatic that verifies that the buffer pointer passed in is not NULL.

Reply:
is there any benifit to add assert to unit test code to test the input but not the result?

Reply:
It looks like there already is a configASSERT( pxEventGroupBuffer ) in xEventGroupCreateStatic.

This configASSERT introduces a decision point / branch when configASSERT is defined, so should be covered with a test case.

Reply:
Added